lib/collection/src/operations/config_diff.rs

```diff
@@ -108,18 +108,25 @@ pub struct OptimizersConfigDiff {
     /// Note: 1Kb = 1 vector of size 256
     #[serde(alias = "max_segment_size_kb")]
     pub max_segment_size: Option<usize>,
-    /// Maximum size (in KiloBytes) of vectors to store in-memory per segment.
+    /// Maximum size (in kilobytes) of vectors to store in-memory per segment.
     /// Segments larger than this threshold will be stored as read-only memmaped file.
-    /// To enable memmap storage, lower the threshold
+    ///
+    /// To enable memmap storage, lower the threshold.
+    /// To disable memmap storage entirely, set to `0`. It will be treated as the largest value possible.
+    ///
     /// Note: 1Kb = 1 vector of size 256
     #[serde(alias = "memmap_threshold_kb")]
     #[validate(range(min = 1000))]
     pub memmap_threshold: Option<usize>,
-    /// Maximum size (in KiloBytes) of vectors allowed for plain index.
-    /// Default value based on <https://github.com/google-research/google-research/blob/master/scann/docs/algorithms.md>
-    /// Note: 1Kb = 1 vector of size 256
+    /// Maximum size (in kilobytes) of vectors allowed for plain index.
+    /// Default value based on <https://github.com/google-research/google-research/blob/master/scann/docs/algorithms.md>.
+    ///
+    /// Recommendend minimum value is `1000`.
+    ///
+    /// When set to `0`, plain index will be disabled. It will be treated as the largest value possible.
+    ///
+    /// Note: 1kB = 1 vector of size 256.
     #[serde(alias = "indexing_threshold_kb")]
-    #[validate(range(min = 1000))]
```

Comment:
Is validation here disabled on purpose? I think we should keep it, but with an updated rule. Also in other places where this PR now disables it.

What we had (1000+), with zero, sounds like a good baseline to me.

We can achieve that with a custom validation rule. I've done this before for some gRPC validations:

- https://github.com/qdrant/qdrant/blob/750cf65bbbad630b5febb91a8d8495c612d57cdf/lib/api/src/grpc/validate.rs#L90-L93
- https://github.com/qdrant/qdrant/blob/750cf65bbbad630b5febb91a8d8495c612d57cdf/lib/api/src/grpc/validate.rs#L125-L156
- https://github.com/qdrant/qdrant/blob/750cf65bbbad630b5febb91a8d8495c612d57cdf/lib/api/src/grpc/qdrant.rs#L131

Reply:
We discussed again, and I think it is better to disable. Not having correct limits in OpenAPI might be confusing, and _maybe_ sometimes lower values are desirable. Previously, we introduced validations because other values causer crashes, but this is just inefficient

Reply:
Makes sense. Thanks for the update. Perfect!